qcodes/dataset/sqlite_base.py

```diff
@@ -1332,7 +1334,16 @@ def get_parameter_data(conn: ConnectionPlus,
         param_names = [param.name for param in paramspecs]
         types = [param.type for param in paramspecs]
 
-        res = get_data(conn, table_name, param_names, start=start, end=end)
+        res = get_parameter_tree_values(conn,
+                                        table_name,
+                                        output_param,
+                                        *param_names[1:])
+
+        start = start or 1
+        end = end or len(res)
```

Comment:
if i remember correctly, `0 or len(res)` returns `len(res)`, hence just simple `or` is not enough for `Optional[int]`. same for `start`.
```suggestion
        end = end if end is not None else len(res)
```

Reply:
It's not the first time you point this out to me. Perhaps I'll eventually learn 🙂